web3/providers/persistent/websocket.py

```diff
@@ -59,18 +59,19 @@ class WebSocketProvider(PersistentConnectionProvider):
     logger = logging.getLogger("web3.providers.WebSocketProvider")
     is_async: bool = True
 
-    _ws: Optional[WebSocketClientProtocol] = None
-
     def __init__(
         self,
         endpoint_uri: Optional[Union[URI, str]] = None,
         websocket_kwargs: Optional[Dict[str, Any]] = None,
         # `PersistentConnectionProvider` kwargs can be passed through
         **kwargs: Any,
     ) -> None:
+        # initialize the endpoint_uri before calling the super constructor
         self.endpoint_uri = (
             URI(endpoint_uri) if endpoint_uri is not None else get_default_endpoint()
         )
+        super().__init__(**kwargs)
```

Comment:
It looks like we do a fair amount of moving around the `super().__init__...` call throughout this PR - I'm curious about why we need to do that now, but not before?

Reply:
I think it's cleaner to understand what arguments need to be instantiated before the `super.__init__()` call and describe them inline. There's some reliance on a few arguments needing to exist beforehand so I moved everything else after this init to make that distinction more clear for a reader.